content/logs/languages/php.md

```diff
@@ -0,0 +1,562 @@
+    // configure your logger as below 
+});
+
+return $app;
+```
+
+#### CakePHP
+
+First, add this dependency to the `composer.json` file and
+run `composer update`.
+
+```json
+{
+    "require": {
+        "cakephp/monolog": "*"
+    }
+}
+```
+
+Then, start by creating a logging configuration file (i.e. `app/Config/log.php`) that you include your `app/Config/bootstrap.php`:
+
+```php
+include 'log.php';
+```
+
+A basic configuration, to replicate what Cake does but using Monolog would look something like this:
+
+```
+CakePlugin::load('Monolog');
+```
+
+Finally log into a file:
+
+```
+CakeLog::config('debug', array(
+  'engine' => 'Monolog.Monolog',
+  'channel' => 'app',
+  'handlers' => array(
+    'Stream' => array(
+      LOGS . 'application-json.log',
+      'formatters' => array(
+        'Json' => array("")
+      )
+    )
+  )
+));
+```
+
+## PHP Symfony
+
+This section is about:
+
+* How to send your logs to Datadog from a Symfony 2 environment
+* How to configure Monolog to send logs in JSON
+* How to enrich your Log events with session contextual data
+
+### Setup - Log into files with Monolog
+
+1. Declare a Monolog JSON formatter as a service:
+
+    ```yaml
+    services:
+        monolog.json_formatter:
+            class: Monolog\Formatter\JsonFormatter
+    ```
+
+2. Wire the formatter in your Monolog configuration: declare the formatter field as follows:
+
+    ```yaml
+     monolog:
+        handlers:
+            main:
+                type:   stream
+                path:   "%kernel.logs_dir%/%kernel.environment%.log"
+                level:  debug
+                formatter: monolog.json_formatter
+    ```
+
+### Configure your Datadog agent
+
+Create a `php.d/conf.yaml` file in your `conf.d/` folder with the following content:
+
+```yaml
+init_config:
+
+instances:
+    
+##Log section
+logs:
+
+    ## - type : file (mandatory) type of log input source (tcp / udp / file)
+    ##   port / path : (mandatory) Set port if type is tcp or udp. Set path if type is file
+    ##   service : (mandatory) name of the service owning the log
+    ##   source : (mandatory) attribute that defines which integration is sending the logs
+    ##   sourcecategory : (optional) Multiple value attribute. Can be used to refine the source attribtue
+    ##   tags: (optional) add tags to each logs collected
+
+  - type: file
+    path: /path/to/your/php/application-json.log
+    service: php
+    source: php
+    sourcecategory: sourcecode
+```
+
+### Adding a session processor to add variable context in your logs
+
+1. Implement your session processor:  
+  This is an example of such processor. It knows the current session and it enriches the content of the log record with valuable information such as the `requestId`, `sessionId`, ...  
+
+    ```php
+     <?php
+
+    namespace Acme\Bundle\MonologBundle\Log;
+
+    use Symfony\Component\HttpFoundation\Session\Session;
+
+    class SessionRequestProcessor
+    {
```

Comment:
In previous PHP code samples, you have the brace on the same line as the `class`/`function` definitions. I would change this code sample to be as follows:

```php
     <?php

    namespace Acme\Bundle\MonologBundle\Log;

    use Symfony\Component\HttpFoundation\Session\Session;

    class SessionRequestProcessor {
      private $session;
      private $sessionId;
      private $requestId;
      private $_server;
      private $_get;
      private $_post;

      public function __construct(Session $session) {
        $this->session = $session;
      }

      public function processRecord(array $record) {
        if (null === $this->requestId) {
          if ('cli' === php_sapi_name()) {
            $this->sessionId = getmypid();
          } else {
            try {
              $this->session->start();
              $this->sessionId = $this->session->getId();
            } catch (\RuntimeException $e) {
              $this->sessionId = '????????';
            }
          }
          $this->requestId = substr(uniqid(), -8);
          $this->_server = array(
            'http.url' => (@$_SERVER['HTTP_HOST']).'/'.(@$_SERVER['REQUEST_URI']),
            'http.method' => @$_SERVER['REQUEST_METHOD'],
            'http.useragent' => @$_SERVER['HTTP_USER_AGENT'],
            'http.referer' => @$_SERVER['HTTP_REFERER'],
            'http.x_forwarded_for' => @$_SERVER['HTTP_X_FORWARDED_FOR']
          );
          $this->_post = $this->clean($_POST);
          $this->_get = $this->clean($_GET);
        }
        $record['http.request_id'] = $this->requestId;
        $record['http.session_id'] = $this->sessionId;
        $record['http.url'] = $this->_server['http.url'];
        $record['http.method'] = $this->_server['http.method'];
        $record['http.useragent'] = $this->_server['http.useragent'];
        $record['http.referer'] = $this->_server['http.referer'];
        $record['http.x_forwarded_for'] = $this->_server['http.x_forwarded_for'];

        return $record;
      }

      protected function clean($array) {
        $toReturn = array();
        foreach(array_keys($array) as $key) {
          if (false !== strpos($key, 'password')) {
            // Do not add
          } else if (false !== strpos($key, 'csrf_token')) {
            // Do not add
          } else {
            $toReturn[$key] = $array[$key];
          }
        }

        return $toReturn;
      }
    }
```